### PR TITLE
fixed a bug where the plugin didn't close correctly

### PR DESCRIPTION
**Pull request by HusseinKabbout**

### CODE REVIEW

File: InstantPrintPlugin.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def initGui(self):
 
         self.toolButton.toggled.connect(self.__enableTool)
         self.iface.mapCanvas().mapToolSet.connect(self.__onToolSet)
-
+
```
Comment (manisandro) [inline InstantPrintPlugin.py]: Whitespace change (decent editors strip trailing whitespaces when saving the file, perhaps eric can be configured to do that)

File: InstantPrintTool.py [hunk 2]
```diff
@@ -16,7 +16,16 @@
 
 from ui.ui_printdialog import Ui_InstantPrintDialog
 
-
+class InstantPrintDialog(QDialog):
+    
+    hidden = pyqtSignal()
+    
+    def __init__(self):
+        QDialog.__init__(self)
```
Comment (manisandro) [inline InstantPrintTool.py]: Change to 

    def __init__(self, parent):
        QDialog.__init__(self, parent)

File: InstantPrintTool.py [hunk 3]
```diff
@@ -28,9 +37,11 @@ def __init__(self, iface, populateCompositionFz=None):
         self.pressPos = None
         self.populateCompositionFz = populateCompositionFz
 
-        self.dialog = QDialog(self.iface.mainWindow())
+        self.dialog = InstantPrintDialog()
+        self.dialog.setWindowFlags(Qt.WindowStaysOnTopHint)
```
Comment (manisandro) [inline InstantPrintTool.py]: Actually it's better to just pass the parent (`self.iface.mainWindow()`) to the `InstantPrintDialog` constructor (as was done with the `QDialog` before), that way the window manager ensures the dialog is above the qgis main window, but not generally on top of *all* windows.

File: InstantPrintTool.py [hunk 4]
```diff
@@ -188,6 +205,7 @@ def canvasReleaseEvent(self, e):
             self.oldrubberband = None
             self.mapitem.setNewExtent(QgsRectangle(self.rect))
 
+
```
Comment (manisandro) [inline InstantPrintTool.py]: Extra whitespace change

File: InstantPrintTool.py (hunk 3 above)
Comment (manisandro) [reply InstantPrintTool.py]: You added the parent argument in InstantPrintDialog which is fine, but you also need to actually pass it here. And remove the WindowStaysOnTopHint flag.

File: InstantPrintPlugin.py [hunk 5]
```diff
@@ -15,7 +15,6 @@
 import os
 
 from InstantPrintTool import InstantPrintTool
-import resources_rc
```
Comment (manisandro) [inline InstantPrintPlugin.py]: Why do you remove this line?

Comment (HusseinKabbout) [reply InstantPrintPlugin.py]: I removed it because it was unused according to pep8.

File: InstantPrintTool.py [hunk 6]
```diff
@@ -28,9 +39,10 @@ def __init__(self, iface, populateCompositionFz=None):
         self.pressPos = None
         self.populateCompositionFz = populateCompositionFz
 
-        self.dialog = QDialog(self.iface.mainWindow())
+        self.dialog = InstantPrintDialog(parent)
```
Comment (manisandro) [inline InstantPrintTool.py]: Please do some minimal tests whether your changes do actually work. `parent` is a non-existing variable and consequently the plugin does not even load.

**Pull request closed without merging**